engine/src/main/java/io/camunda/zeebe/engine/state/ZbColumnFamilies.java

```diff
@@ -7,100 +7,115 @@
+  DEFAULT(0),
 
   // util
-  KEY,
+  KEY(1),
 
   // process
-  PROCESS_VERSION,
+  PROCESS_VERSION(2),
 
   // process cache
-  PROCESS_CACHE,
-  PROCESS_CACHE_BY_ID_AND_VERSION,
-  PROCESS_CACHE_DIGEST_BY_ID,
+  PROCESS_CACHE(3),
+  PROCESS_CACHE_BY_ID_AND_VERSION(4),
+  PROCESS_CACHE_DIGEST_BY_ID(5),
 
   // element instance
-  ELEMENT_INSTANCE_PARENT_CHILD,
-  ELEMENT_INSTANCE_KEY,
+  ELEMENT_INSTANCE_PARENT_CHILD(6),
+  ELEMENT_INSTANCE_KEY(7),
 
-  NUMBER_OF_TAKEN_SEQUENCE_FLOWS,
+  NUMBER_OF_TAKEN_SEQUENCE_FLOWS(8),
 
   // variable state
-  ELEMENT_INSTANCE_CHILD_PARENT,
-  VARIABLES,
+  ELEMENT_INSTANCE_CHILD_PARENT(9),
+  VARIABLES(10),
   @Deprecated
-  TEMPORARY_VARIABLE_STORE,
+  TEMPORARY_VARIABLE_STORE(11),
 
   // timer state
-  TIMERS,
-  TIMER_DUE_DATES,
+  TIMERS(12),
+  TIMER_DUE_DATES(13),
 
   // pending deployments
-  PENDING_DEPLOYMENT,
-  DEPLOYMENT_RAW,
+  PENDING_DEPLOYMENT(14),
+  DEPLOYMENT_RAW(15),
 
   // jobs
-  JOBS,
-  JOB_STATES,
-  JOB_DEADLINES,
-  JOB_ACTIVATABLE,
+  JOBS(16),
+  JOB_STATES(17),
+  JOB_DEADLINES(18),
+  JOB_ACTIVATABLE(19),
 
   // message
-  MESSAGE_KEY,
-  MESSAGES,
-  MESSAGE_DEADLINES,
-  MESSAGE_IDS,
-  MESSAGE_CORRELATED,
-  MESSAGE_PROCESSES_ACTIVE_BY_CORRELATION_KEY,
-  MESSAGE_PROCESS_INSTANCE_CORRELATION_KEYS,
+  MESSAGE_KEY(20),
+  MESSAGES(21),
+  MESSAGE_DEADLINES(22),
+  MESSAGE_IDS(23),
+  MESSAGE_CORRELATED(24),
+  MESSAGE_PROCESSES_ACTIVE_BY_CORRELATION_KEY(25),
+  MESSAGE_PROCESS_INSTANCE_CORRELATION_KEYS(26),
 
   // message subscription
-  MESSAGE_SUBSCRIPTION_BY_KEY,
+  MESSAGE_SUBSCRIPTION_BY_KEY(27),
   @Deprecated // only used for migration logic
-  MESSAGE_SUBSCRIPTION_BY_SENT_TIME,
+  MESSAGE_SUBSCRIPTION_BY_SENT_TIME(28),
   // migration end
-  MESSAGE_SUBSCRIPTION_BY_NAME_AND_CORRELATION_KEY,
+  MESSAGE_SUBSCRIPTION_BY_NAME_AND_CORRELATION_KEY(29),
 
   // message start event subscription
-  MESSAGE_START_EVENT_SUBSCRIPTION_BY_NAME_AND_KEY,
-  MESSAGE_START_EVENT_SUBSCRIPTION_BY_KEY_AND_NAME,
+  MESSAGE_START_EVENT_SUBSCRIPTION_BY_NAME_AND_KEY(30),
+  MESSAGE_START_EVENT_SUBSCRIPTION_BY_KEY_AND_NAME(31),
 
   // process message subscription
-  PROCESS_SUBSCRIPTION_BY_KEY,
+  PROCESS_SUBSCRIPTION_BY_KEY(32),
   // migration start
   @Deprecated // only used for migration logic
-  PROCESS_SUBSCRIPTION_BY_SENT_TIME,
+  PROCESS_SUBSCRIPTION_BY_SENT_TIME(33),
   // migration end
 
   // incident
-  INCIDENTS,
-  INCIDENT_PROCESS_INSTANCES,
-  INCIDENT_JOBS,
+  INCIDENTS(34),
+  INCIDENT_PROCESS_INSTANCES(35),
+  INCIDENT_JOBS(36),
 
   // event
-  EVENT_SCOPE,
-  EVENT_TRIGGER,
+  EVENT_SCOPE(37),
+  EVENT_TRIGGER(38),
+
+  BANNED_INSTANCE(39),
+
+  EXPORTER(40),
+
+  AWAIT_WORKLOW_RESULT(41),
+
+  JOB_BACKOFF(42),
 
-  BANNED_INSTANCE,
+  DMN_DECISIONS(43),
+  DMN_DECISION_REQUIREMENTS(44),
+  DMN_LATEST_DECISION_BY_ID(45),
+  DMN_LATEST_DECISION_REQUIREMENTS_BY_ID(46),
+  DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY(47),
 
-  EXPORTER,
```

Comment:
Aligning the CF definitions is something that comes when we backport #16072 to 8.1.

This PR is focused on avoiding more problems in the future, as well as adding some safety when backporting #16072 to 8.1.

@Zelldon Was your expectation that this PR already re-aligns the CFs? In that case, this PR also has to recover the lost data right? Otherwise, we're in an in-between state where we have lost the data on 8.1

Reply:
✍️  For completeness, we discussed this here: https://github.com/camunda/zeebe/pull/16129#discussion_r1472809402